[JSON] add ActivateScreenSaver

DaVukovic opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
This changes introduces the option to activate the currently used screensaver. 
It's already available as a Python built-in and I thought it might be a good idea to have that as well. 

If this is not the correct way to implement that, then please let me know how to do it correct. For example, I'm not so sure if the version number I`m using is correct of if it should be 12.5.0
So please let me know if anything is wrong or could be done better.

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
solves: #20088

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
I did a few tests with Kodi on Linux and also tried with an additionally installed screensaver add-on (bing fotos of the week). 

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->
At least nothing negative ;). Users are simply able to activate the screensaver via a JSON call. It doesn't break anythin. 

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Tolriq commented:
Should be 12.5.0 as API change :) Beware that as soon as there's JSON divergence with Matrix future JSON patch won't be able to be backported so maybe it should be backported too as simple to keep backports opened.

About "namespace" is this really System and not more GUI? I suppose this would be a no-op on a future GUI less Kodi. (The built in does not have System.ActivateScreenSaver it's just labelled as that in the Wiki.

BTW if you are in adding the missing stuff here's the biggest one that can't be used from JSON and still require Event Server to run the built in.
```
PlayerControl(tempoup)
PlayerControl(tempodown)

CECActivateSource
CECStandby
CECToggleState

PlayDisc
RipCD
```

garbear commented:
BTW, the commit without whitespace changes is easier to review, here: https://github.com/xbmc/xbmc/commit/9b9db66af648e79dd807ca515a00b31a7105764b?w=1 (appended `?w=1` to raw commit)

DaVukovic commented:
@Tolriq 
Thanks for your input

>About "namespace" is this really System and not more GUI? I suppose this would be a no-op on a future GUI less Kodi. (The built in does not have System.ActivateScreenSaver it's just labelled as that in the Wiki.

True. My bad. I thought one needs to trigger it via `System.ActivateScreenSaver`. But `ActivateScreenSaver` is what one needs to call. So I will change that to GUI-Operations. 

Will do a backport of that for sure if it gets accepted. 

@garbear 
>BTW, the commit without whitespace changes is easier to review, ....

That was a mistake by me. I force-pushed without commiting ;). Hence the commit. Please ignore that. I'll do an update of this PR anyway.

garbear commented:
> That was a mistake by me. I force-pushed without commiting ;). Hence the commit. Please ignore that. I'll do an update of this PR anyway.

If you are a git-fu master, you'll do a "cleanup" commit first and a "meaty" commit after, so that the meaty commit has exactly what needs to be reviewed ;)

DaVukovic commented:
> If you are a git-fu master

Well, I'm not ;)

I've just pushed the latest changes. If this PR is hard to review, I'll happily close it and create a new one.

garbear reviewed (approved):
[Link to new commit](https://github.com/xbmc/xbmc/pull/20110/commits/e05f9f13f6e4c6b0d8cd676aea1a6e9a77607998). I'm not sure about the version number, if you should bump the minor or the patch. But otherwise this looks like a clean addition of a feature that makes sense.

Montellese reviewed (commented):
Looks good apart from the unnecessary whitespace changes in `JSONServiceDescription.cpp`.

DaVukovic commented:
Yeah, that's what Jenkins suggests

lrusak commented:
@DaVukovic would it be possible for you to split out the clang-format changes for `CJSONServiceDescription::m_methodMaps` into a pre commit? This way your commit changes exactly what is needed.

DaveTBlake reviewed (changes requested):
Unhappy with having whitespace changes throughout JSONServiceDescription.cpp in this PR. 
Since it is part of our coding standard to allow additional whitespace in code for readability of things like data structures, I would suggest that the necessary `// clang-format on` `// clang-format off` are added for this instead of stripping whitespace. Otherwise the addition of the method is fine.  

Note merging this will mark the end of fixes to JSON API in Matrix. There are several other API changes waiting to merge that have been delayed for this reason. Perhaps we should merge them in order of creation rather than give this priority? The only reason to do this one first would be if we also backport it to v19? Any opinions?

howie-f commented on xbmc/interfaces/json-rpc/JSONServiceDescription.cpp:
```diff
@@ -38,6 +38,7 @@ std::map<std::string, JSONSchemaTypeDefinitionPtr> CJSONServiceDescription::m_ty
 CJSONServiceDescription::IncompleteSchemaDefinitionMap CJSONServiceDescription::m_incompleteDefinitions = CJSONServiceDescription::IncompleteSchemaDefinitionMap();
 
 JsonRpcMethodMap CJSONServiceDescription::m_methodMaps[] = {
+// clang-format off
```
only need to move this line above `JsonRpcMethodMap…` and we‘re done. thanks

DaVukovic commented:
@howie-f 

Obviously something got totally mixed up as I tried to squash commits and revert old clang-stuff. I hope I've done everything correct now. At least it builds fine for me and I can use this method.

howie-f reviewed (approved):
nice 👍

Commit:
[cmake] introduce PLATFORM_OPTIONAL_DEPS_EXCLUDE

Allow a platform to exclude a dependency from the optional_deps list.
This example includes CEC that isnt used for ios/tvos/android/UWP

Commit:
[cmake] Create P8-Platform find module

Apply patches for p8-platform to fix potential cmake install location issues

Commit:
[cmake] implement ENABLE_INTERNAL_CEC

Allows building libcec (and dependency p8-platform) as part of core cmake project.

Commit:
[cmake] module helper allow override of byproduct extension

Allows a lib to override the extension of a byproduct. This is used for example by
macos when a shared lib is used. The extension in this case is dylib, but on other unix
platforms its so

joseluismarti committed:
[Android] Check if dovi profile is supported by the codec

rmrector committed:
add thumb loader for pvr channel groups

rmrector committed:
just set pvr channel group thumb path in thumbloader

rmrector committed:
remove old tiled-image generator

that saved directly to the texture cache.

CrystalP committed:
[uPnP] Fix playback from authenticated sources

Credentials must be added to the URL for sources that require authentication.

CrystalP committed:
[Windows] videosync: retrieve refresh rate only on display config change.

The previous condition was true for every frame and retrieved the current refresh rate too often.

IsCurrent() also reacts to display changes other than refresh rate but display changes are rare.
IsCurrent() takes about 1/10th of the time of refresh rate retrieval and mostly saves time

CrystalP committed:
[Windows] Handle short WaitForVBlank

This happens when closing the lid of a laptop, during refresh rate/HDR status change and for errors.
Before attempting to detect another vblank, sleep to avoid a busy loop.
Since the user can't see the output, no need to attempt to maintain perfect vsync with an expensive
accurate alternative source such as spinning on performance counters.
It's better to not send updates to the video reference clock and let it interpolate from the last
known good timestamp)

CrystalP committed:
[Windows] Retrieve the screen refresh rate without using the swap chain

The video clock runs on its own thread.
Previous implementation with DeviceResources::GetDisplayMode() used a swap chain method that sometimes
gets stalled by an internal dxgi lock held by the main thread calling Present(). The additional
information is not even useful here => skip

notspiff committed:
remove unused method

notspiff committed:
remove unused headers

notspiff committed:
remove custom deleters

these were rendered unnecessary when destructor was
moved to translation unit in eecc6665f34e8445d7b1f548b8be6fd74167fb54

icr8ions committed:
* fix python XBMCAddon::xbmcgui::Window::onAction() action.getButtonCode()

* changes to populate buttonCode for CAction such that it is available for python addons with custom controllers

ksooo committed:
[PVR] CGUIWindowPVRBase: Prevent concurrent updates caused by GUI_MSG_UPDATE.

HiassofT committed:
keymaps: change remote poweroff action to show shutdown menu

Signed-off-by: Matthias Reichl <hias@horus.com>

DaVukovic committed:
[Eventlog] fix eventlog message

DaVukovic commented:
Missed that one totally. I'll have a look soon and will probably rebase it. If not, I'll close this one

CrystalP committed:
[Windows] const-ify system clock frequency

CrystalP committed:
[Windows] Add GetOutput() alternative, using cached information first.

Windowing events already maintain an up to date dxgi output member. Make it track the output
description as well, and use those members from the vsync tracking rather than spending resources
retrieving the current dxgi output every frame.

Other benefits:
- avoids dxgi synchronization with main thread Present() for GetContainingOutput() and output->GetDesc().
- avoids GetContainingOutput() errors when the swapchain doesn't own the output, for example with
WARP device and sometimes with multi-screen UWP desktop that can't catch some events due to UWP limitations.

Commit:
[cmake] Modulehelper allow overriding CMAKE_INSTALL_LIBDIR

joseluismarti committed:
FFmpeg: flag --enable-neon has no actual effects on Android targets

Commit:
[cmake] enable copy_file_to_buildtree to handle files created at build time

This allows files in DEPENDS_PATH to be handled at build time to install to
the target build tree. This is required when for example a shared lib is built at build time
and needs to be bundled. The existing globbing wont find the lib at generation time,
therefore it doesnt get bundled.

sundermann committed:
[webOS] Pause video on minimize

sundermann committed:
Remove GLES3/gl3ext.h include

This file an empty header that is only available in mesa. It's not part of the original khronos GLES3 headers

lrusak committed:
CNetworkLinux: implement ping using sockets instead of system call to ping

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
[TEST] CNetworkTest: add basic test for PingHost

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

Commit:
[cmake] Create FindPythonInterpreter for host information

This splits out host find information from the regular target find module.
This allows us to not mix and check for host executable status, and only explicitly
search for the host interpreter when required (eg ENABLE_EVENTCLIENTS=ON)

joseluismarti committed:
[Android] Clang -march flag is not needed

a1rwulf committed:
Properly add application data to the websocket pong

The RFC states that pong should reply with the same application
data that is received in the ping.
Our code however threw away the application data instead.
We need to properly set application data length so that it actually
gets appended to the send buffer.

sundermann committed:
[webOS] Renderer: Fix render region

lrusak committed:
CPlatformFreebsd: use app params to set audio backend

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
[windows] add winsock2.h to PlatformDefs.h

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

CrystalP committed:
[Windows] Remove unnecessary sleep for half vsync

Windows 7 is not supported anymore in v21 and the sleep was necessary only for Windows 7 as back to
back WaitForVBlank calls would otherwise freeze rendering.

Sleep is not needed anymore with Windows 8 and later, and detrimental with UWP, which doesn't have an
API to increase timer precision and frequently overslept.

lrusak committed:
CCompileInfo: add ability to get enabled gl interfaces

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
CAppParamParserLinux: add --gl-interface=<interface> switch

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
CWinSystemX11GLContext: select gl interface from command line switch

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
kodi.sh.in: add wrapper for deprecated KODI_GL_INTERFACE env variable

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

ksooo committed:
[utils] CGUIContentUtils::HasInfoForItem: Let PVR check the item. Do not make assumptions about PVR's implementation of OnInfo.

jjd-uk committed:
[GUI][Skins] Bump to xbmc.gui 5.17.0

ksooo committed:
[fileitem][PVR] CFileItem::LoadDetails: Factor out loading of PVR items.

ksooo committed:
[PVR] Add support for channels to CPVRGUIActionsUtils::LoadItem.

ksooo committed:
[application] Fix bd selection menu shown when external player is used.

Commit:
[Omega] Beta1

ksooo committed:
[PVR] Fix 'switch to previous channel' not working if it is in another group than the currently playing channel.

enen92 committed:
[wayland][input] Remove dead code

ksooo committed:
[fileitem] Fix mimetype not set/updated properly on loading details.

enen92 committed:
[wayland][input] Add xkb Compose and dead-keys support

Rechi committed:
[clang-tidy] performance-faster-string-find

Rechi committed:
[clang-tidy] performance-unnecessary-copy-initialization

Rechi committed:
[clang-tidy] performance-no-automatic-move

enen92 committed:
[input] Add dead keys to xbmc keysim definitions

enen92 committed:
[wayland][input] Save unicode translation of the pressed deadkey

enen92 committed:
[GUIEditControl] Allow the cursor character to be defined. Allow blinking behavior to be set

enen92 committed:
[input][inputmanager] Add keyboard composing event and action

enen92 committed:
[input][wayland] Notify outside world about keyboard composing events

ksooo committed:
[favourites] CFavouritesService: Add missing locks.

a1rwulf committed:
Fix websocket pong and probably other messages

When "handle" gets called it already creates a perfectly
fine "answer"-frame.
Calling the Send method of CWebSocketClient will wrap it into
another TextFrame which may result in an  1007 Illegal UTF-8 Sequence
or possibly other invalid websocket data frames.
We need to call Send of CTCPClient instead which will just transfer
the frame without further modification.

enen92 committed:
[GUIEditControl] Modify cursor behavior if a key is composing

enen92 committed:
[input][wayland] Allow replay sequence on compose key cancellation

CrystalP committed:
[rendering] Create CToneMappers to share luminance calculation across platforms

CrystalP committed:
[rendering] use default luminance for tone mapping of streams with bad metadata

ksooo committed:
[favourites] Fix FAVOURITES_UTILS::MoveItem to ensure item gets not destructed during vector content adjustment.

ksooo committed:
[favourites] Fix CFavouriteContextMenuAction::Execute to return false if DoExecute fails.

enen92 committed:
[wayland] Add logger to libxkbcommon

enen92 committed:
[wayland][input] Make key composition optional

Commit:
[tvos] Remove Boblight

Boblight isnt currently packaged, its only for a single platform, and it no longer builds
with xcode 15. Just remove it

Commit:
[darwin_embedded] Silence GL deprecation warnings

Xcode 15 has a define GLES_SILENCE_DEPRECATION to silence all the GL deprecation
warnings. No idea if its in 14 or not, but it'll have zero effect on xcode
versions that dont support that define.

Commit:
[TVOS] increase min SDK version to TVOS12.0

Xcode 15 no longer supports 11.0. Increase minimum to 12.0

Commit:
[tvos] Fix some errors in copy scripts

DllPaths_generated.h only has .so libs, so change from dylib to match
We currently have no dylibs in $TARGET_FRAMEWORKS/"*dylib*, so just remove
this block. Fixes this message

error: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: /Users/Shared/jenkins/workspace/TVOS/build/build/Debug-appletvos/Kodi.app/Frameworks/*dylib* (No such file or directory)

CrystalP committed:
[Rendering] Share heuristics to guess primaries across platforms

Common implementation in VideoPicture for code that was copy/paste in DX/GL/GLES
New member of VideoPicture to keep the original primaries

ksooo committed:
[keymaps] Remove leftovers from favourites dialog.

CrystalP committed:
[Windows] Display original / effective primaries in Video Debug OSD

sundermann committed:
[webOS] Platform: Fix HOME variable append

Commit:
[cmake] FindPulseAudio fix compile definition

rmrector committed:
explicitly note when hash is revalidated

ksooo committed:
[playlistplayer] Fix PVR items not handled correctly. Those need to be played using CApplication::PlayMedia

ksooo committed:
[cores] Fix CDVDInputStreamBluray::Open check for item being resumable.

ksooo committed:
[video] CGUIWindowVideoBase::OnItemInfo: Skip check whether a video info tag can be filled in case we already have one.

Commit:
[cmake] FindFMT further simplification

ksooo committed:
[PVR] Implement ListItem.IsPlaying for channels so that same channel contained in different groups is correctly handled.

ksooo committed:
[PVR] Fix CPVRGUIActionsChannels::GetSelectedChannelPath to respect the currently active channel group.

ksooo committed:
Revert "[playlistplayer] Fix PVR items not handled correctly. Those need to be played using CApplication::PlayMedia"

This reverts commit 3816a0ada63c824b82c1404df17a4b813256b727.

ksooo committed:
[interfaces][json-rpc] CPlayerOperations::Open : Add support for PVR channels and recordings handed in via 'file' property.

thexai committed:
Fix Visual Studio warnings

ksooo committed:
[PVR] CPVRPlaybackState : Improve handling of hidden and deleted groups.

joseluismarti committed:
[Android] Change the directory used to verify the SDK path

garbear committed:
Format GUIControlFactory.cpp/h

garbear committed:
Update break statements in GUIControlFactory.cpp

ksooo committed:
[video] Fix CVideoDatabase::GetMusicVideosByWhere to set dyn path for loaded items (like we do in all the other Get*ByWhere methods already).

Commit:
[tools/depends] Bump wayland 1.22.0

Commit:
[tools/depends][target] Bump wayland-protocols 1.32

Commit:
[tools/depends] Bump WaylandPP 1.0.0

Commit:
[tools/depends][target] Add ZLIB dependency to mesa target

smp79 committed:
VDPAU: fix hq scalers

ksooo committed:
[video] Fix 'Play using' offering RetroPlayer for playback of iso bd/dvd disc images.

ksooo committed:
[PVR] PVRDatabase: Store and retrieve id of the group a channel was last watched from.

ksooo committed:
[PVR] Improve recently played channels widget (e.g. respect hidden groups and channels).

xodidox committed:
Added Video Versions feature

garbear committed:
Fix controller not rendering for copied gamecontroller controls

When a gamecontroller control is in a list-within-a-list with static
listitems provided by a <content> node, the controller image can
disappear after the inner list is updated.

The reason is because the control path for copying lists-within-lists
does not properly clear m_currentController, causing the static state
to not set the filename because the new controller has the same ID as
m_currentController.

However, CGUIImage (which gamecontroller derives from) properly resets
image filename state when copied, so skip the check for the existing
controller ID and let CGUIImage handle filenames that don't change.

CrystalP committed:
[Windows] Disable video settings not supported by the DXVA processor.

Brightness and contrast video settings were unconditionally enabled in Windows.
Fixed by propagating the brightness, contrast and rotation capabilities of the dxva processor to
the video settings screen.
Rotation needs more work to enabled/disable according to renderer caps.

Commit:
[cmake] set EXCLUDE_FROM_ALL property for build_internal_depends target

We need to set this proeprty so we dont inadvertently build all internal dependencies
regardless of state due to the ALL target

Dandan336 committed:
Add LoongArch support to system info

sundermann committed:
[depends][pycryptodome] Set pycryptodome library suffix

Commit:
[tools/depends] Bump Python 3.11.6

smp79 committed:
VAAPI: don't flush ffmpeg buffers if it holds any refs on close

smp79 committed:
VDPAU: don't flush ffmpeg buffers if it holds any refs on close

smp79 committed:
DXVA: don't flush ffmpeg buffers if it holds any refs on close

Commit:
[tools/depends] Bump Openssl1.1.1w

Commit:
[cmake] add ${CORE_PLATFORM_NAME_LC}_SEARCH_CONFIG NO_DEFAULT_PATH for windows

ksooo committed:
[docs] Changed sample code to use prefix operator in for loops intstead of postfix operator.

ksooo committed:
[docs] Make clear, that the guidelines should be followed also for changes in existing files.

Hitcher committed:
Fix home categories focus position

This fixes the problem where static content gets focused first in mixed content lists by simply moving it in front pf the dynamic content. Problem was raised on the forum - https://forum.kodi.tv/showthread.php?tid=374267

lrusak committed:
CEGLFence: add support for using EGL_KHR_fence_sync and EGL_ANDROID_native_fence_sync

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
CDRMAtomic: add support for using DRM_MODE_ATOMIC_NONBLOCK

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

ksooo committed:
[docs] Coding guideline: Add, that we are using prefix operators in traditional for loop's increment statement.

CastagnaIT committed:
[python][window] Delete pcontrol after SendGUIMessage

rmrector committed:
if file is empty return empty full-path

rmrector committed:
CVideoGeneratedImageFileLoader check setting to extract thumb

enen92 committed:
Revert "[upnp] Fix smart playlist folder definitions"

This reverts commit b3542a899eb1670de91190403f6174a6b309d6b7.

enen92 committed:
[upnp] SmartPlaylists are also folders/containers

sundermann committed:
[depends][PIL] Set library suffix instead of loader patching

mglae committed:
Fix xmbc.getRegion("time")

- Always convert minutes and seconds, fix after #22982
- Complete time formatting codes from https://kodi.wiki/view/Language_support#langinfo.xml

ksooo committed:
[PVR] const correctness improvements.

mglae committed:
Update doxygen of xmbc.getRegion()

FergusGeorge committed:
Estuary OSD showing wrong value for ‘Next’ programme when playing radio. 

Change the order of the tests used to establish what type of media is playing when determining how to set OSDNextLabelVar. 

Variables.xml currently sets the value of OSDNextLabelVar by checking if the ‘Visualisation” window is active before checking whether we’re playing ‘livetv’. 

Any radio shows played through the PVR pass the ‘livetv’ test, and can correctly show the Start Time, End Time and Name of the next programme.  This currently fails however, because the ‘Visualisation’ check is successfully satisfied first.

lrusak committed:
[linux][docs] add note about libdisplay-info when building kodi with gbm windowing

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

thexai committed:
[FileSystem] Add GUI settings for NFS and SMB chunk size

Hitcher committed:
Fix Shift view for Collections

Hitcher committed:
Revert "Fix Shift view for Collections"

This reverts commit 740b9c3c9b9b9a2473fc7f35b4fffd0dd4d26eba.

ksooo committed:
[video][music][interfaces] Fix playback of strm library items (e.g. from Estuary home screen widgets).

neo1973 committed:
Picture: Switch internal functions to use smart pointers

This fixes #23993.

rmrector committed:
Invert logic

ksooo committed:
[fileitem] Fix CFileItem::IsFileFolder to not treat strm files as browsable folders.

rmrector committed:
Update xbmc/TextureCache.cpp

Co-authored-by: Frank H. <58829855+howie-f@users.noreply.github.com>

Hitcher committed:
Fix 'Shift' view for Collections

neo1973 committed:
Picture: Remove unnecessary nullptr checks

ksooo committed:
[PVR] const correctness improvements, part 2.

ksooo committed:
[PVR] const correctness improvements, part 3.

ksooo committed:
[PVR] const correctness improvements, part 4.

ksooo committed:
[PVR] const correctness improvements, part 5.

ksooo committed:
[video] Video info dialog's play button now respects default play action setting.

ksooo committed:
[PVR] Recording info dialog's play button now respects default play action setting.

DeltaMikeCharlie committed:
Add locale InfoLabels for Skins

ksooo committed:
[PVR] Guide info dialog's play button now respects default play action setting.

kambala-decapitator committed:
migrate from altool to notarytool

enen92 committed:
[libinput] Add logger to keyboard implementation

enen92 committed:
[libinput] use smart pointers with custom deleters to wrap xkb raw pointers

enen92 committed:
[libinput] Add support for dead-keys/xkb-compose

enen92 committed:
[libinput] Replace std::map with utils/map

enen92 committed:
[libinput] Use unsigned int for repeat constexprs

ksooo committed:
[contextmenumanager] Fix 'More...' context menu item displayed for favourites, although no target context menu items are available.

rmrector committed:
use default member initialization for whole class

kambala-decapitator committed:
use keychain profile for notarytool

ksooo committed:
[builtins] PlayerBuiltins: PlayOrQueueMedia: Fix playback of PVR items (for example from Estuary home screen).

ksooo committed:
[builtins] PlayerBuiltins: PlayOrQueueMedia: Fix playlist ids.

ksooo committed:
[builtins] PlayerBuiltins: PlayOrQueueMedia: Fix resume dialog presented although default play/select action is configured not to ask.

ksooo committed:
[fileitem] Fix 'action play' not working for favourites for strm files.

ksooo committed:
[contextmenumanager][favourites][listproviders] Get rid off CContextMenuManager default parameters.

ksooo committed:
[contextmenumanager] Add/complete doxy for the public interface of CContextMenuManager.

Hitcher committed:
Fix alignment of counter labels

kambala-decapitator committed:
use new variables in scripts

enen92 committed:
[xkbcommon] Unify xkbLogger

thexai committed:
[FileSystem] Move File Cache settings to GUI settings

smp79 committed:
VAAPI: increase maxReferences for AV1

enen92 committed:
[CApp] Kill XBMC to Kodi migration

ksooo committed:
[PVR] Fix recordings with a path containing a ':' not displayed in recordings window.

ksooo committed:
[fileitem] CFileItem::LoadDetails: Add support for strm files containing an audio item.

ksooo committed:
[PVR][Estuary] PVR core should not set label2 for PVR recording folder items. This should be done by the skins, the same way as already implemented for tv shows, seasons, movie sets.

garbear committed:
[Doxy] Improve documentation of "gamewindow" control

garbear committed:
[Doxy] Improve documentation of "gamecontroller" and "gamecontrollerlist"

Commit:
[cmake] copy_file_to_buildtree make sure we create ExportFiles.cmake

thexai committed:
Fix crash due nullptr surface using DXVA2

This may happen when a stream starts with non I frame.

thexai committed:
DVDInputStreamBluray: Fix missing file flags on creation

thexai committed:
DVDInputStreamBluray: remove unnecessary initialization of smart pointer

the-black-eagle committed:
[python] Add songVideoURL to InfoTagMusic

garbear committed:
[Doxy] Update Window IDs

garbear committed:
[Doxy] Improve formatting of RetroPlayer.* built-in options

garbear committed:
Update strings.po with translations for game window IDs

CrystalP committed:
[videoplayer] Modify Backwards Chapter Skip

After a grace delay, a backwards chapter skip goes to the beginning of the current chapter instead of skipping to the beginning of the previous chapter.

ksooo committed:
[video] GUIWindowVideoNav.cpp: Remove unused include.

garbear committed:
[Doxy] Fix formatting of Window IDs table

garbear committed:
[Estuary] Give example static content for skin devs

garbear committed:
[Doxy] Add game-related files to doxy group

garbear committed:
[Doxy] Improve documentation of new GameAgents window

garbear committed:
[Doxy] Improve documentation of the controller configuration dialog

garbear committed:
[Doxy] Improve documentation of the port setup dialog

thexai committed:
[Xbox] Add support for HDR10 passthrough

fritsch committed:
ffmpeg: Bump to version 6.0.1

This fixes #23198
Changelog: https://git.ffmpeg.org/gitweb/ffmpeg.git/shortlog/n6.0.1

lrusak committed:
tools/depends: curl: format configure options

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-zstd option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

razzeee committed:
Remove article from summary

lrusak committed:
tools/depends: curl: add --without-brotli option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-gssapi option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-gsasl option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-hyper option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-ngtcp2 option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-nghttp3 option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-quiche option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-msh3 option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-gnutls option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-nss option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-mbedtls option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --without-wolfssl option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

lrusak committed:
tools/depends: curl: add --with-zlib option

Signed-off-by: Lukas Rusak <lorusak@gmail.com>

Hitcher committed:
Fix weather layout when busy

Hitcher committed:
Revert "Backport weather fix"

Commit:
[cmake] FindEGL set CMAKE_REQUIRED_INCLUDES for check_include_files

CrystalP committed:
[Settings] Adjust the "Use display HDR capabilities" setting

Revised name and description.

joseluismarti committed:
[Android] Unused variable m_colorFormat in MediaCodec

thexai committed:
[Xbox] Fix crash on close App

garbear committed:
Add tranlation for Peripherals dialog with ID 10149

Before:

  <visible>Window.IsVisible(10149)</visible>

After:

  <visible>Window.IsVisible(Peripherals)</visible>

ksooo committed:
[builtins] PlayerBuiltins::PlayOrQueueMedia: Fix logging to only log an error in case the item cannot be played/queued. Not being able to get playlist items is not always an error.

ksooo committed:
[builtins] PlayerBuiltins::PlayOrQueueMedia: Exit early if no items to play/queue.

ksooo committed:
[PVR] Fix racing problems due to ACTION_CHANNEL_SWITCH message being posted instead of sent (which was recently introduced, but seems to have bad side effects).

rmrector committed:
avoid creating a var just to return value

thexai committed:
[VideoPlayer] Fix cache level calculation and add to Debug Info OSD

ksooo committed:
[video] CAsyncItemsForPlaylist: If we shall resume but can't, default to play from beginning.

CrystalP committed:
[Settings][Windows] Adjust description of DXVA VSR setting

CrystalP committed:
[Estuary] Modify settings decription field to avoid autoscroll

Was problematic for resolutions < 1080p

Rechi committed:
[clang-tidy] modernize-make-unique

Rechi committed:
[clang-tidy] modernize-use-emplace

Rechi committed:
[clang-tidy] performance-no-automatic-move

thexai committed:
[VideoPlayer] Remove unused code

thexai committed:
[VideoPlayer] Update cache-related variables descriptions

78andyp committed:
Correctly handle 'Disc n' subdirectories in scraping and set creation.

the-black-eagle committed:
Cleanup DefaultVideoVersions icon

xodidox committed:
[VideoVersion] Added VideoVersionPlay dialog

ksooo committed:
[video] Fix auto play next item not working if 'uncategorized' is not selected in the respective GUI setting.

enen92 committed:
[media] Consistency accross platforms splashscreens (avoid versions)

CrystalP committed:
[cleanup] remove a few warnings

CrystalP committed:
[cleanup] remove unused variable

thexai committed:
Fix Visual Studio warnings

Commit:
[swig] Fix misleading indentation warnings

Example of warning fixed by specifically encasing the else
/build/build/swig/AddonModuleXbmcdrm.i.cpp:183:7: warning: misleading indentation; statement is not part of the previous 'else' [-Wmisleading-indentation]
      if (pymimeType) PyXBMCGetUnicodeString(mimeType,pymimeType,false,"mimeType","GetKeyRequest");
      ^
/build/build/swig/AddonModuleXbmcdrm.i.cpp:181:5: note: previous statement is here
     else
     ^

neo1973 committed:
XBMCTinyXML2: Strip trailing '\0' when writing to file

Fixes #24135.

quietvoid committed:
[Android] Set codec profile 8 in DOVI compatibility mode

garbear committed:
[Games] Remove hard-coded button combos

CrystalP committed:
[Windows] Address recommendations of VS code analysis tools for WinMain

annotations of WinMain
return code of CoInitializeEx and WSAStartup not checked

CrystalP committed:
[Windows] Close the application mutex when exiting

ReleaseMutex was incorrect because the mutex ownership is not requested and it's not waited on.
Instead the handle should be closed.

joseluismarti committed:
[Estuary] Add Theora video codec flag image

xodidox committed:
[VideoVersion] Fixed unexpected VideoVersion dialog in VideoInfo dialog when resuming playback

CrystalP committed:
[Windows] Uninitialize properly for all exit paths

CrystalP committed:
[Windows] Extract command line parsing into its own function

jjd-uk committed:
Estuary - add media flags for music

Commit:
[metadata.generic.artists] 1.0.18

jjd-uk committed:
[Estuary) Video Version skin redesign + cleanup

[Estuary) Video Version skin redesign + cleanup

quietvoid committed:
[Android] Ignore obsolete DOVI profiles from codec selection

When the AVDOVIDecoderConfigurationRecord is not valid, the fields are zeroed
and profile is set to 1, which is incorrect. It should default to profile 0.

Fixes #24171

CrystalP committed:
[Estuary] fullscreen info: expand the plot textbox for item by item playback

Instead of leaving a blank when playing single files, use the space used by the label displaying the next item to add an extra line to the plot.

CrystalP committed:
[Estuary] fullscreen info: hide next item label for last playlist item
